pkg/window/strategy/fixed/lifecycle.go

```diff
@@ -0,0 +1,130 @@
+package fixed
+
+import (
+	"container/list"
+	"sync"
+	"time"
+
+	"github.com/numaproj/numaflow/pkg/window"
+	"github.com/numaproj/numaflow/pkg/window/keyed"
+)
+
+// ActiveWindows maintains the state of active windows
+// All the operations in ActiveWindows order the entries in the ascending order of start time.
+// So the earliest window is at the front and the oldest window is at the end.
+type ActiveWindows struct {
+	// entries is the list of active windows that are currently being tracked.
+	// windows are sorted in chronological order with the earliest window at the head of the list.
+	// list.List is implemented as a doubly linked list which allows us to traverse the nodes in
+	// both the directions.
+	// Although the worst case time complexity is O(n), because of the time based ordering and
+	// since the elements are rarely out of order, the amortized complexity works out to be closer to O(1)
+	// Because most of the keys are expected to be associated with the most recent window, we always start
+	// the traversal from the tail of the list for Get and Create Operations. For Remove Operations, since
+	// the earlier windows are expected to be closed before the more recent ones, we start the traversal
+	// from the Head.
+	entries *list.List
+	lock    sync.RWMutex
+}
+
+// NewWindows initializes the windows object.
+func NewWindows() *ActiveWindows {
+	return &ActiveWindows{
+		entries: list.New(),
+		lock:    sync.RWMutex{},
+	}
+}
+
+// CreateKeyedWindow adds a new keyed window for a given interval window
+func (aw *ActiveWindows) CreateKeyedWindow(iw *window.IntervalWindow) *keyed.KeyedWindow {
+	aw.lock.Lock()
+	defer aw.lock.Unlock()
+
+	kw := keyed.NewKeyedWindow(iw)
+
+	// this could be the first window
+	if aw.entries.Len() == 0 {
+		aw.entries.PushFront(kw)
+		return kw
+	}
+
+	earliestWindow := aw.entries.Front().Value.(*keyed.KeyedWindow)
+	recentWindow := aw.entries.Back().Value.(*keyed.KeyedWindow)
+
+	// late arrival
+	if earliestWindow.Start.Equal(kw.End) || earliestWindow.Start.After(kw.End) {
+		aw.entries.PushFront(kw)
+	} else if recentWindow.End.Equal(kw.Start) || recentWindow.End.Before(kw.Start) {
```

Comment:
User `after`?

Reply:
`kw.Start.After(recentWindow.End)`?  if so, its the same. are you suggesting that writing like this may be more readable?

Reply:
Similarly here.

Reply:
right

Reply:
made the change.